src/components/data/PublicLinks.js

```diff
@@ -0,0 +1,247 @@
+import { ERROR_CODES, getErrorCode } from "components/utils/error/errorCode";
+import GridLoading from "components/utils/GridLoading";
+import ErrorTypographyWithDialog from "components/utils/error/ErrorTypographyWithDialog";
+
+import {
+    Button,
+    Hidden,
+    Paper,
+    TextField,
+    Toolbar,
+    Typography,
+    useTheme,
+} from "@material-ui/core";
+
+import { makeStyles } from "@material-ui/core/styles";
+import { Skeleton } from "@material-ui/lab";
+
+import {
+    CheckCircle,
+    CloudDownload,
+    Error,
+    Save,
+    FileCopy,
+} from "@material-ui/icons";
+
+const useStyles = makeStyles((theme) => ({
+    toolbarButtons: {
+        flexGrow: 1,
+        margin: theme.spacing(0.5),
+    },
+    paper: {
+        padding: theme.spacing(0.5),
+    },
+}));
+
+function PublicLinks(props) {
+    const classes = useStyles();
+    const { baseId, paths } = props;
+    const theme = useTheme();
+    const [links, setLinks] = useState();
+    const [download, setDownload] = useState(false);
+    const [saveAsDialogOpen, setSaveAsDialogOpen] = useState(false);
+    const [saveNewFileError, setSaveNewFileError] = useState(null);
+    const [fileSavePath, setFileSavePath] = useState();
+    const [successMsg, setSuccessMsg] = useState();
+    const [errorMsg, setErrorMsg] = useState();
+    const { t } = useTranslation("data");
+
+    const { isFetching, error } = useQuery({
+        queryKey: [PUBLIC_LINKS_QUERY_KEY, paths],
+        queryFn: getPublicLinks,
+        config: {
+            enabled: paths && paths.length > 0,
+            onSuccess: (resp) => {
+                const pathsWithLink = resp?.paths;
+                let parsedLinks = "";
+                paths.forEach((path) => {
+                    parsedLinks = parsedLinks.concat(
+                        `${pathsWithLink[path]}\n`
+                    );
+                });
+                setLinks(parsedLinks);
+            },
+        },
+    });
+
+    const [saveTextAsFile, { status: fileSaveStatus }] = useMutation(
+        uploadTextAsFile,
+        {
+            onSuccess: (resp) => {
+                setSuccessMsg(
+                    t("fileSaveSuccess", {
+                        fileName: resp?.file.label,
+                    })
+                );
+                setSaveAsDialogOpen(false);
+            },
+            onError: (error) => {
+                const text =
+                    getErrorCode(error) === ERROR_CODES.ERR_EXISTS
+                        ? t("fileExists", {
+                              path: getParentPath(fileSavePath),
+                          })
+                        : t("fileSaveError");
+                setSaveNewFileError(text);
+            },
+        }
+    );
+
+    useEffect(() => {
+        if (download) {
+            window.open(
+                `${getHost()}/api/downloadText?text=${encodeURIComponent(
+                    links
+                )}`,
+                "_blank"
+            );
```

Comment:
I may be missing something, but it seems like instead of using the `download` state and useEffect, this could just be a normal function that gets called when the `Download` button gets clicked?

Reply:
`window` can be `undefined` when use outside `useEffect` since Nextjs does SSR. `useEffect` is guaranteed to run on the browser. Same logic is used in File Viewers `Toolbar.js` as well. Please see for https://dev.to/vvo/how-to-solve-window-is-not-defined-errors-in-react-and-next-js-5f97 for more info.

Reply:
Got it - thanks for the link!